### ISSUE TITLE
Remove Elastic Agent V1 control protocol

### ISSUE THREAD

**Issue by jlind23** [elastic/elastic-agent-client#32]
Now that the V2 protocol is out from 8.3 on, the V1 will not be updated and thus removed in a few releases. (Around end of 2022)

**Comment by jlind23** [comment]
@pierrehilbert this is something you should read and take into consideration for the next upcoming releases.

**Comment by michalpristas** [comment]
who;s in charge of `Change all beats to start without reading a configuration file from the disk (preferably they should start, context to control protocol and wait for initial configuration)` ?

**Comment by cmacknz** [comment]
Yes, we can help with changing how the beats start up under agent. None of us are particularly familiar with how beats run under agent yet though, so if you can send us a detailed issue about what to change it would be greatly appreciated. 

The same for `Change beats to understand configuration directly from policy (all rules are removed in the V2 binary specification)`. I know we are relying on rules for filebeat and metricbeat but we didn't write the spec files originally.

**Comment by michalpristas** [comment]
i can write something up about how config is loaded now and what needs to be changed tomorrow

**Comment by michalpristas** [comment]
@cmacknz 
## Beats behavior change

### Current
At the moment we handle process as this:
- elastic agent starts
- elastic agent is notified about or detects config change
- elastic agent creates a specific change for specific beats (let's assume filebeat from now on)
- elastic agent starts filebeat process with `-management.enabled` flag
- filebeat process starts
- filebeat loads a config from `filebeat.yml`
```golang
	cfg, err := cfgfile.Load("", settings.ConfigOverrides)
	if err != nil {
		return fmt.Errorf("error loading config file: %w", err)
	}
```
- filebeat initializes manager to get comms with agent going 
```golang
	// initialize config manager
	b.Manager, err = management.Factory(b.Config.Management)(b.Config.Management, reload.Register, b.Beat.Info.ID)
	if err != nil {
		return err
	}
```
- filebeat fetches config from fleet

### Proposed
- elastic agent starts
- elastic agent is notified about or detects config change
- elastic agent creates a specific change for specific beats (let's assume filebeat from now on)
- elastic agent starts filebeat process with `-management.enabled` flag
- filebeat process starts
- in case `management.enabled` flag is not set filebeat loads a config from `filebeat.yml` otherwise use sane default configuration
```golang
	cfg, err := cfgfile.Load("", settings.ConfigOverrides)
	if err != nil {
		return fmt.Errorf("error loading config file: %w", err)
	}
```
- filebeat initializes manager to get comms with agent going 
```golang
	// initialize config manager
	b.Manager, err = management.Factory(b.Config.Management)(b.Config.Management, reload.Register, b.Beat.Info.ID)
	if err != nil {
		return err
	}
```
- filebeat fetches config from fleet

**Comment by cmacknz** [comment]
Thanks! I've created two issues to track the needed work in beats and updated this issue description to track them:

- https://github.com/elastic/beats/issues/31901
- https://github.com/elastic/beats/issues/31900

For migrating to the V2 specification file/removing config transformations I've only considered filebeat and metricbeat since those are the only beats we own. They are also probably the most work to port.

**Comment by cmacknz** [comment]
Issue to migrate libbeat to the V2 protocol: https://github.com/elastic/beats/issues/32078
